src/lib/utils/dataUtils.ts

```diff
@@ -7,22 +7,46 @@ export function renderItem<T extends Element = HTMLElement>(
   data: any,
   append: boolean = false,
 ): T {
-  return render<T>(container, append, () => template(data))[0];
+  return render<T>(container, append, () => createHtml(template(data)))[0];
 }
 
 export function renderItems<T extends Element = HTMLElement>(
   container: HTMLElement,
   template: (data?: any) => string,
   data: Array<any>,
   append: boolean = false,
+  listElementWrapper?: HTMLElement,
 ) {
-  return render<T>(container, append, () => data.reduce((html, d) => html + template(d), ''));
+  let generateStringTemplate;
+  if (listElementWrapper) {
+    generateStringTemplate = () => {
+      return data.reduce((html, d) => {
+        const templateHtml = createHtml(template(d), <HTMLElement>listElementWrapper.cloneNode(
+          true,
```

Comment:
Looking at this `deep` clone, it got me thinking that we don't even support Nested wrappers.

If we pass `<div><span></span></div>`, we'll append our rendered template to the outer div (that's the reference we pass), not to the inner span (what we might want).
So for this to work we need to clone the outer element deep, but in the createHtml function, we need to search for the deepest child element to set the innerHTML.

Getting more and more complex :D

Reply:
Haha whoops left the `clone` on that line by accident

Getting more complex but also more interesting :)